lib/features/modeling/util/ModelingUtil.js

```diff
@@ -37,19 +42,48 @@ export function getParent(element, anyType) {
  * Determines if the local modeling direction is vertical or horizontal.
  *
  * @param {Element} element
+ * @param {ElementRegistry} elementRegistry
  *
  * @return {boolean} false for vertical pools, lanes and their children. true otherwise
  */
-export function isDirectionHorizontal(element) {
+export function isDirectionHorizontal(element, elementRegistry = undefined) {
+
+  var parent = getParent(element, 'bpmn:Process');
+  if (parent) {
+    return true;
+  }
 
   var types = [ 'bpmn:Participant', 'bpmn:Lane' ];
 
-  var parent = getParent(element, types);
+  parent = getParent(element, types);
   if (parent) {
     return isHorizontal(parent);
-  } else if (isAny(element, types)) {
+  } else if (is(element, types)) {
     return isHorizontal(element);
   }
 
-  return true;
+  var process;
+  for (process = getBusinessObject(element); process; process = process.$parent) {
+    if (is(process, 'bpmn:Process')) {
+      break;
+    }
+  }
+
+  if (elementRegistry === undefined) {
```

Comment:
Let's use idiomatic JS:

```suggestion
  if (!elementRegistry) {
```